main.go

```diff
@@ -56,16 +58,28 @@ func main() {
 		return
 	}
 
+	appCtx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
 	shutdown := make(chan int)
 	go func() {
 		<-shutdown
+		cancel()
 		wal.Shutdown()
 		os.Exit(0)
 	}()
 
-	win, appWindow, err := ui.CreateWindow(wal)
+	dbPath := filepath.Join(cfg.HomeDir, cfg.Network, "dexc.db")
+	logMaker := initDexLogging(cfg.DebugLevel, true, logWriter{})
+	dexc, err := dexc.NewDex(dbPath, cfg.Network, logMaker)
```

Comment:
Kindly confirm why we are initializing an app dex db. Why isn't this handled from dcrlibwallet?.
Same with the logger.

Reply:
Do you mean like a `dcrlibwallet.InitDexClient` function/method? `dexc.NewDex` ultimately calls `decred.org/dcrdex/client/core.New`, which requires a dbPath and logger though, so regardless of where `core.New` is called from, a separate db will be created by `core` and a dbPath will need to be provided.

Reply:
So basically, when adding dex to mobile, another db will have to be initialized on mobile. I believe we can make this work from dcrlibwallet so we dont have to repeat this on all the platforms we will be integrating dex on.

Reply:
I understand now. That's possible. Rule of thumb will be moving all code bits that don't concern the UI to dcrlibwallet. I'll discuss this with @song50119.